tests/webserver.py

```diff
@@ -0,0 +1,42 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import threading
+
+try:
+    from http.server import HTTPServer, SimpleHTTPRequestHandler
+except ImportError:
+    from BaseHTTPServer import HTTPServer
+    from SimpleHTTPServer import SimpleHTTPRequestHandler
+
+
+class WebServer(object):
+
+    def __init__(self, host='', port=8000):
+        self.server = HTTPServer((host, port), SimpleHTTPRequestHandler)
+        self.thread = threading.Thread(target=self._start_server)
+        self.thread.daemon = True
+
+    def _start_server(self):
+        os.chdir(os.path.join(os.path.dirname(__file__), 'web'))
```

Comment:
You probably want to override `translate_path` on `SimpleHTTPRequestHandler` instead of changing the CWD.

(Again, I think you’d be better off using wptserve for this.)